packages/amplify-prompts/src/validators.ts

```diff
@@ -17,6 +17,11 @@ export const alphanumeric =
   (input: string) =>
     /^[a-zA-Z0-9]+$/.test(input) ? true : message;
 
+export const matchRegex =
+  (validatorRegex : RegExp, message?: string): Validator =>
+  (input: string) =>
+    validatorRegex.test(input) ? true : message || `Input does not match the regular expression ${validatorRegex}`;
```

Comment:
`test()` returns a boolean, so the ternary here is redundant. However, it looks like that is how the surrounding code is written, so no need to update this.